Library Catalog (DRA).js

```diff
@@ -9,80 +9,195 @@
 	"inRepository": true,
 	"translatorType": 4,
 	"browserSupport": "gcsbv",
-	"lastUpdated": "2014-08-26 03:57:40"
+	"lastUpdated": "2016-12-04 12:47:41"
 }
 
-/* No more libraries with permalinks that I know of
-sample URLs: http://libraries.nc-pals.org
-http://web2.libraries.vermont.gov/web2/tramp2.exe/log_in?SETTING_KEY=English
+/*
+Sample URLs: 
+  http://libraries.nc-pals.org --> not working anymore
+  http://web2.libraries.vermont.gov/web2/tramp2.exe/log_in?SETTING_KEY=English
+  http://catalogues.toulouse.fr/web2/tramp2.exe/log_in?setting_key=BMT1
+  https://sdb-web2.biblio.usherbrooke.ca/web2/tramp2.exe/log_in?setting_key=french
  */
 
 function detectWeb(doc, url) {
-	if(doc.location.href.search(/\/authority_hits|\/form\//) > 0) {
-		return "multiple";
-	} else {
+	if (url.indexOf('/see_record/')>-1 || url.indexOf('/goto/')>-1 || url.indexOf('/do_keyword_search/')>-1 || url.indexOf('/authority_hits/')>-1 || url.indexOf('/form/')>-1) {
 		return "book";
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
 	}
 }
 
-function doWeb(doc, url) {
-	var checkItems = false;
-	if(detectWeb(doc, url)== "multiple") {
-		checkItems = Zotero.Utilities.gatherElementsOnXPath(doc, doc, "//ol//tr/td|//ol/li//ul/li", null);
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//ol//tr/td|//ol/li//ul/li');
+	for (var i=0; i<rows.length; i++) {
+		var href = ZU.xpathText(rows[i], './/a[contains(@href, "/web2/tramp2.exe/goto") or contains(@href, "/web2/tramp2.exe/see_record")]/@href');
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
 	}
-	
-	if(checkItems && checkItems.length) {
-		var items = Zotero.Utilities.getItemArray(doc, checkItems, 'https?://.*/web2/tramp2\.exe/(goto|see\_record)');
-		uris=[];
-		Zotero.selectItems(items, function (items) {
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
 				return true;
 			}
+			var articles = [];
 			for (var i in items) {
-				uris.push(i);
+				articles.push(i);
 			}
-			scrape(uris)
-			});
-
+			ZU.processDocuments(articles, scrape);
+		});
 	} else {
-		if (url.indexOf("/do_keyword_search/")!=-1){
-			url = "http://" + doc.location.host + ZU.xpathText(doc, '//td[@class="enrichcontent"]/a[contains(@href, "MARCRecord")]/@href');
+		var marclink = ZU.xpathText(doc, '//td[@class="enrichcontent"]/a[contains(@href, "MARCRecord")]/@href');
+		var metalink = ZU.xpathText(doc, '//meta[@http-equiv="REFRESH"]/@content');
+		if (url.indexOf("/do_keyword_search/")!=-1 || url.indexOf("/form/")>-1) {
+			//Here we need first to go to another website which will then also
+			//create a session to continue.
+			if (marclink) {
+				url = "http://" + doc.location.host + marclink;
```

Comment:
any reason you're using hardcoded http here and agnostic // in line 70 below? the latter seems more robust.